server/dump.go

```diff
@@ -90,3 +98,52 @@ func (p *Plugin) createDump(ctx context.Context, min, max time.Time, remoteStora
 
 	return zipFileNameRemote, nil
 }
+
+// Metadata is the auxiliary data added to a plugin downloadable.
+// It should contain the required fields and may have it's custom values as well.
+// TODO: move this to mattermost/server/public/pluginapi in the future
+type metadata struct {
+	ServerVersion string         `yaml:"server_version"`
+	ServerID      string         `yaml:"server_id"`
+	LicenseID     string         `yaml:"license_id"`
+	PluginID      string         `yaml:"plugin_id"`
+	Custom        map[string]any `yaml:"custom"`
+}
+
+func (p *Plugin) genarateMetadataForDump(path string, min, max time.Time) (*metadata, error) {
+	m := &metadata{}
+	if l := p.API.GetLicense(); l != nil {
+		m.LicenseID = l.Id
+	}
+
+	m.ServerVersion = p.API.GetServerVersion()
+	m.ServerID = p.API.GetTelemetryId()
+	m.PluginID = root.Manifest.Id
+
+	// Add plugin specific metadata
+	customMetadata := map[string]any{
+		"generated": time.Now().Format(time.RFC822),
+		"min":       min.UnixMilli(),
+		"max":       max.UnixMilli(),
+	}
+
+	m.Custom = customMetadata
+
+	b, err := yaml.Marshal(m)
```

Comment:
I picked `yaml` to make it a bit more human-readable. `json` is also fine for me.

Reply:
3/5 both are fine for this purpose. It would be nice if the file type is consistent across the different types of packages.